### PR TITLE
loop: Loop In MuSig2 support

### PR DESCRIPTION
**Pull request by bhandras**
This PR defaults new swaps to MuSig 1.0rc and adds support for MuSig loopin swaps for the success case.

Depends on: 
~- https://github.com/lightninglabs/loop/pull/546~

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: swapserverrpc/server.proto [hunk 1]
```diff
@@ -196,14 +199,21 @@ message ServerLoopOutTerms {
 
 message ServerLoopInRequest {
     bytes sender_key = 1;
+
+    bytes sender_internal_key = 9;
```
Comment (sputn1ck) [inline swapserverrpc/server.proto:205]: nit: explicitely state if this is pub or privkey

File: swapserverrpc/server.proto [hunk 2]
```diff
@@ -218,6 +228,9 @@ message ServerLoopInRequest {
 
 message ServerLoopInResponse {
     bytes receiver_key = 1;
+
+    bytes receiver_internal_key = 9;
```
Comment (sputn1ck) [inline swapserverrpc/server.proto:234]: nit: explicitely state if this is pub or privkey

File: loopdb/store_test.go [hunk 3]
```diff
@@ -29,6 +30,26 @@ var (
 		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3,
 	}
 
+	senderInternalKey = [33]byte{
```
Comment (sputn1ck) [inline loopdb/store_test.go:33]: just as a comment: I think we ought to clearly specify if we mean pubkey or privkey, as here it is pubkey, where as in the bolt db bucket it ist the privkey. I would vouch to always specify wether pubkey or privkey is meant

**Review by sputn1ck — COMMENTED** [review]
Very nice PR, easy to grok changes 💯

File: swapserverrpc/server.proto (hunk 2 above)
Comment (bhandras) [reply swapserverrpc/server.proto]: done

File: loopdb/store.go [hunk 4]
```diff
@@ -112,9 +112,49 @@ var (
...
+	// value: serialized public key.
+	receiverInternalPubKeyKey = []byte("receiver-internal-pubkey")
+
+	// senderInternalKeyKey is the key that stores the sender's internal
+	// private key corresponding to the sender's internal public key.
+	//
+	// path: loopInBucket/loopOutBucket -> swapBucket[hash]
+	//                                     -> senderInternalKeyKey
+	// value: serialized private key.
+	senderInternalKeyKey = []byte("sender-internal-key")
```
Comment (sputn1ck) [inline loopdb/store.go:137]: nit: rename to `senderInternalPrivKeyKey`

File: server_mock_test.go [hunk 5]
```diff
@@ -151,17 +151,22 @@ func getInvoice(hash lntypes.Hash, amt btcutil.Amount, memo string) (string, err
...
 
 	_, receiverKey := test.CreateKey(101)
+	_, receiverInternalKey := test.CreateKey(102)
 
 	if amount != s.expectedSwapAmt {
 		return nil, errors.New("unexpected test swap amount")
 	}
 
-	var receiverKeyArray [33]byte
+	var receiverKeyArray, receriverInternalKeyArray [33]byte
```
Comment (sputn1ck) [inline server_mock_test.go:164]: nit: typo `receriverInternalKeyArray`

**Review by sputn1ck — APPROVED** [review]
This is not a draft anymore right? LGTM 🐐

File: server_mock_test.go (hunk 5 above)
Comment (bhandras) [reply server_mock_test.go]: done

File: loopdb/store.go (hunk 4 above)
Comment (bhandras) [reply loopdb/store.go]: done

**Comment by bhandras** [comment]
Added shared key encryption to the stored priv keys.

File: loopin.go [hunk 6]
```diff
@@ -189,6 +190,24 @@ func newLoopInSwap(globalCtx context.Context, cfg *swapConfig,
 		return nil, err
 	}
 
+	// Default the HTLC internal key to our sender key.
+	senderInternalPubKey := senderKey
+	var senderInternalPrivKey [32]byte
+
+	// If this is a MuSig2 swap then we'll generate a brand new key pair
+	// and will use that as the internal key for the HTLC.
+	if loopdb.CurrentProtocolVersion() >= loopdb.ProtocolVersionMuSig2 {
```
Comment (GeorgeTsagk) [inline loopin.go:199]: What does the `>=` imply for the order of protocol versions? Anything that comes after `ProtocolVersionMuSig2 ` is gonna share the same code in the block below? If yes, why?

**Review by GeorgeTsagk — APPROVED** [review]
➰LGTM➰  

minor comments

File: loopin.go (hunk 6 above)
Comment (bhandras) [reply loopin.go:199]: This just means that currently we make the assumption that all further protocol upgrades will include musig2 loopin. If this is not the case we'll simply change it.

File: loopdb/store.go [hunk 7]
```diff
@@ -435,19 +596,10 @@ func (s *boltSwapStore) CreateLoopOut(hash lntypes.Hash,
...
-			)
-			if err != nil {
-				return err
-			}
-
-			err = swapBucket.Put(keyLocatorKey, keyLocator)
-			if err != nil {
-				return err
-			}
+		// Store the htlc keys if the contract is recent.
```
[the remark this answers is not in the document]
Comment (bhandras) [reply loopdb/store.go]: ptal

**Comment by bhandras** [comment]
ping @sputn1ck

**Pull request merged**